### ISSUE TITLE
Multiple secondary emails can be added following the same flow at the same time from different devices

### ISSUE THREAD

**Issue by vasilicatamas** [mozilla/fxa#4546]
**Build:**
Stage [Train 1.162.0]

**Affected platforms:**
Windows 10 x64
Android 9

**Steps to reproduce:**
1. Create and confirm a new Firefox account.
2. Login using the new Firefox account on 2 devices.
3. On Device 1, go to the email inbox and open the "Set up recovery email" email.
4. On Device 2, go to the email inbox and open the "Set up recovery email" email.
5. On Device 2, add a secondary email and click the “Continue” button.
6. On Device 1, add a different secondary email and click the “Continue” button.
7. Confirm both secondary emails.
8. Check in Firefox Account Settings page the emails listed under “Secondary email” section.

**Expected results:**
Only one email can be assigned as a secondary email.

**Actual Results:**
Multiple emails can be set as secondary email following the “Set up recovery email” flow.

![2secondaryemails](https://user-images.githubusercontent.com/18114210/76628764-eeb1cd00-6545-11ea-951a-f3d07e2ddab0.gif)

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-1375)
┆Issue Number: FXA-1375
┆Attachments: <a href="https://jira.mozilla.com/secure/attachment/12807/Screen+Shot+2020-03-23+at+5.49.25+PM.png">Screen Shot 2020-03-23 at 5.49.25 PM.png</a>

**Comment by imskr** [comment]
I would like to fix this error if no one is assigned. @vasilicatamas

**Comment by data-sync-user** [comment]
➤ Wil Clouser commented:

[~adavis@mozilla.com] is supporting multiple secondary emails a bug?

**Comment by data-sync-user** [comment]
➤ Alex Davis commented:

It is not a bug. We can keep this one open.
This should probably be an epic to some extent.

Note that our backend supports having multiple emails but we just don't support it on the front end.

**Comment by data-sync-user** [comment]
➤ Jody Heavener commented:

[~adavis@mozilla.com] does this need to go to UX first, or can I use best judgement here? I'm interested in updating the frontend to address this.

**Comment by data-sync-user** [comment]
➤ Alex Davis commented:

[~jheavener@mozilla.com] I'm pretty flexible but it will also depend on what you have in mind.

Is your goal to surface the ability to add multiple emails in fxa preferences to align with this edge case flow? Or is it to prevent people from encountering the flow described above?

TBH, I'm surprised our UI actually displayed all the multiple secondary emails but the good news is that I think it means we're more than halfway to officially supporting it.

**Comment by data-sync-user** [comment]
➤ Jody Heavener commented:

[~adavis@mozilla.com] I was thinking since we support it we may as well surface it in the FxA settings. I don't imagine it would need to be too clever; just an additional button to trigger the New email flow. Maybe some tweaks to ensure each email is rendering with enough space. That "Remove" button is a little in your face when there are multiple email rows, so we could move them to a ghost-style button, or maybe switched the text out for an X icon, but I don't think this is critical.

I've attached a really simple example to this ticket

**Comment by data-sync-user** [comment]
➤ Alex Davis commented:

> That "Remove" button is a little in your face when there are multiple email rows

I agree but let's leave the big red buttons for now. We can address design within the context of the greater redesign that is happening in parallel.

**Comment by vasilicatamas** [comment]
Verified this fix on FxA Stage [Train 1.165.0] under Windows 10 64bit, Android 9 and iOS 13.1.2. One new issue was found during this testing #4751 and considering that it is tracked in a separate bug I will mark this one as Verified.

![Screenshot_25](https://user-images.githubusercontent.com/18114210/78367787-4f737900-75cb-11ea-8b16-acc58436fb15.png)
